src/main/java/org/spongepowered/api/text/sink/PermissionSink.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.sink;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.command.CommandSource;
+import org.spongepowered.api.service.permission.PermissionService;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class PermissionSink implements MessageSink {
+
+    protected final String permission;
+
+    public PermissionSink(String permission) {
+        this.permission = checkNotNull(permission, "permission");
+    }
+
+    @Override
+    public Iterable<CommandSource> getRecipients() {
+        PermissionService service = Sponge.getGame().getServiceManager().provideUnchecked(PermissionService.class);
+
+        return service.getKnownSubjects().values().stream()
+                .flatMap(input -> input.getAllWithPermission(this.permission).entrySet().stream()
+                        .filter(Map.Entry::getValue)
+                        .map(entry -> entry.getKey().getCommandSource().orElse(null))
+                        .filter(source -> source != null))
+                .collect(Collectors.toList());
```

Comment:
This should be a `Set` to avoid duplicate entries and the order is really not important here.